lib/dotcom_web/controllers/trip_plan_controller.ex

```diff
@@ -5,305 +5,43 @@ defmodule DotcomWeb.TripPlanController do
-      destination = %NamedPosition{
-        latitude: String.to_float(latitude),
-        longitude: String.to_float(longitude),
-        name: name,
-        stop: nil
-      }
+  @doc """
+  When visiting /trip-planner/:from/some%20address or /trip-planner/:to/some%20address,
+  we lookup the location and redirect to the trip planner with that location encoded in the query string.
+  """
+  def location(conn, %{"direction" => direction, "query" => query})
+      when direction in ["from", "to"] do
 
-      do_from(conn, destination)
-    else
-      updated_address = check_address(address)
+    case @location_service.geocode(query) do
+      {:ok, [%LocationService.Address{} = location | _]} ->
+        encoded = build_params(direction, location) |> AntiCorruptionLayer.encode()
+        path = live_path(conn, DotcomWeb.Live.TripPlanner)
 
-      case @location_service.geocode(updated_address) do
-        {:ok, [geocoded_from | _]} ->
-          do_from(conn, NamedPosition.new(geocoded_from))
+        redirect(conn, to: "#{path}?plan=#{encoded}") |> halt()
 
-        _ ->
-          # redirect to the initial index page
-          redirect(conn, to: trip_plan_path(conn, :index))
-      end
+      _ ->
+        location(conn, %{})
     end
   end
 
-  defp do_from(conn, destination) do
-    # build a default query with a pre-filled 'from' field:
-    query = %Query{
-      from: destination,
-      to: {:error, :unknown},
-      time: {:error, :unknown}
-    }
-
-    now = Util.now()
-
-    # build map information for a single leg with the 'from' field:
-    map_data =
-      TripPlanMap.itinerary_map([
-        %Leg{
-          from: destination,
-          to: nil,
-          mode: %PersonalDetail{},
-          start: now,
-          stop: now
-        }
-      ])
-
-    %{markers: [marker]} = map_data
-    from_marker = %{marker | id: "B"}
-    map_info_for_from_destination = %{map_data | markers: [from_marker]}
-
-    conn
-    |> assign(:query, query)
-    |> assign(:map_data, map_info_for_from_destination)
-    |> render(:index)
-  end
+  def location(conn, _params) do
+    path = live_path(conn, DotcomWeb.Live.TripPlanner)
 
-  def to(conn, %{"plan" => _plan} = params) do
-    redirect(conn, to: trip_plan_path(conn, :index, Map.delete(params, "address")))
+    redirect(conn, to: path) |> halt()
   end
 
-  def to(conn, %{
-        "address" => address
-      }) do
-    if String.match?(address, ~r/^(\-?\d+(\.\d+)?),(\-?\d+(\.\d+)?),.*$/) do
-      [latitude, longitude, name] = String.split(address, ",", parts: 3)
-      # Avoid extra geocode call, just use these coordinates
-      destination = %NamedPosition{
-        latitude: String.to_float(latitude),
-        longitude: String.to_float(longitude),
-        name: name,
-        stop: nil
+  defp build_params(direction, location) do
+    %{
+      "#{direction}" => %{
+        "latitude" => location.latitude,
+        "longitude" => location.longitude
```

Comment:
oh, I think if you just add `"name" => query` here that might fix the input

Reply:
yeah, i didn't know what that was for. i'll add it in.